editor/src/clj/editor/collection_non_editable.clj

```diff
@@ -263,31 +265,87 @@
 
 (g/defnk produce-node-outline [_node-id]
   {:node-id _node-id
-   :node-outline-key "Collection Data"
-   :label "Collection Data"
+   :node-outline-key "Non-Editable Collection"
+   :label "Non-Editable Collection"
    :icon collection-common/collection-icon})
 
-(g/defnk produce-scene [_node-id]
-  {:node-id _node-id
-   :aabb geom/null-aabb})
+(defn- make-desc->instance-scene [node-id desc->source-scene child-id->desc]
+  {:pre [(ifn? desc->source-scene)
+         (ifn? child-id->desc)]}
+  (letfn [(desc->instance-scene [desc]
+            (let [id (:id desc)
+                  transform (any-instance-desc->transform-matrix desc)
+                  source-scene (desc->source-scene desc)
+                  instance-scene (collection-common/any-instance-scene node-id id transform source-scene)
+                  child-instance-scenes (map (comp desc->instance-scene child-id->desc)
+                                             (:children desc))]
+              (cond-> instance-scene
+                      (seq child-instance-scenes)
+                      (update :children util/intov child-instance-scenes))))]
+    desc->instance-scene))
+
+(g/defnk produce-scene [_node-id collection-desc embedded-component-resource-data->scene-index embedded-component-scenes referenced-component-proj-path->scene-index referenced-component-scenes referenced-collection-proj-path->index referenced-collection-scenes referenced-game-object-proj-path->index referenced-game-object-scenes]
+  (let [prototype-desc->scene
+        (game-object-non-editable/make-prototype-desc->scene embedded-component-resource-data->scene-index embedded-component-scenes referenced-component-proj-path->scene-index referenced-component-scenes)
+
+        any-game-object-instance-desc->source-scene
+        (fn any-game-object-instance-desc->source-scene [game-object-instance-desc]
+          ;; GameObject$InstanceDesc or GameObject$EmbeddedInstanceDesc in map format.
+          (if-some [referenced-game-object-proj-path (:prototype game-object-instance-desc)]
+            (-> referenced-game-object-proj-path
+                referenced-game-object-proj-path->index
+                referenced-game-object-scenes)
+            (-> game-object-instance-desc
+                :data
+                (prototype-desc->scene _node-id))))
+
+        collection-instance-desc->source-scene
+        #(-> % :collection referenced-collection-proj-path->index referenced-collection-scenes)
+
+        game-object-instance-descs
+        (concat (:embedded-instances collection-desc)
+                (:instances collection-desc))
+
+        [game-object-instance-id->game-object-instance-desc child-game-object-instance-id?]
+        (util/into-multiple [{} #{}]
+                            [(map (juxt :id identity))
+                             (mapcat :children)]
+                            game-object-instance-descs)
```

Comment:
I've added `into-multiple` for when you want to produce multiple outputs from a single pass over a sequence. Doing it in one pass instead of multiple helps keep the CPU cache misses down, and this results in a one-third to one-half of the runtime of doing multiple passes.